tsdb/engine/tsm1/wal.go

```diff
@@ -399,6 +401,9 @@ func (l *WAL) writeToLog(entry WALEntry) (int, error) {
 	// limit how many concurrent encodings can be in flight.  Since we can only
 	// write one at a time to disk, a slow disk can cause the allocations below
 	// to increase quickly.  If we're backed up, wait until others have completed.
+	l.limiter.Take()
+	defer l.limiter.Release()
```

Comment:
Hmmm it looks like these statements were originally in the code, then commented out in d7c5dd0a3e2bdcaa669d2d7ece8bd76ef07b075a and eventually removed. Putting them back makes sense to me given the comment, but I'll need to call in another reviewer with more context.

@StoneYunZhao have you seen memory pressure in this code that's been fixed by your changes here? Or did you notice the comment and impl didn't align while reading through the code?

Reply:
My situation belongs to the latter.

Reply:
When there is a peak write traffic, I see that the memory will increase a lot. But I am not sure if it is caused by this code, It may be `Cache` in `tsdb/engine/tsm1/cache.go`, or it may be both.

Reply:
> Hmmm it looks like these statements were originally in the code, then commented out in [d7c5dd0](https://github.com/influxdata/influxdb/commit/d7c5dd0a3e2bdcaa669d2d7ece8bd76ef07b075a) and eventually removed. Putting them back makes sense to me given the comment, but I'll need to call in another reviewer with more context.
> 
> @StoneYunZhao have you seen memory pressure in this code that's been fixed by your changes here? Or did you notice the comment and impl didn't align while reading through the code?

I think the purpose of https://github.com/influxdata/influxdb/commit/d7c5dd0a3e2bdcaa669d2d7ece8bd76ef07b075a is to reduce the frequency of goroutine creation and destruction, but I don't understand why the limiter related code is deleted.